/*
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 *
 * TestService.proto
 * Defines a simple service for testing the RpcChannel
 * Copyright (C) 2005 Simon Newton
 */

package ola.rpc;

option cc_generic_services = false;
option py_generic_services = true;

message EchoRequest {
  required string data = 1;
  optional int64 session_ptr = 2;
}

message EchoReply {
  required string data = 1;
}

message STREAMING_NO_RESPONSE {

}

service TestService {
  rpc Echo (EchoRequest) returns (EchoReply);
  rpc FailedEcho (EchoRequest) returns (EchoReply);
  rpc Stream (EchoRequest) returns (STREAMING_NO_RESPONSE);
}
